### PR TITLE
Update service.py

### PR DESCRIPTION
**Pull request by lj894**
Bug fix to remove serialize() call.
Removed helper class and did code clean up to match the style of other functions.

### CODE REVIEW

**Comment by sushanthsamala** [comment]
@lj894 Since, you opened a new pull request for this task and my latest changes are merged. I would suggest you to add in unit tests for the end point that you added and the related model.py functions. Currently there is no check if "id" is int or not a negative number or not etc.

**Comment by MinervaMint** [comment]
I think the `find` method in model.py may need to be modified. Now it uses `cls.query.get(by_id)` which did not work when I was running it locally - I think it is bacause that `id` is not the only primary key, and maybe `cls.query.filter` is what is needed?

**Comment by lj894** [comment]
Updated the code and added the test cases. Please review again.

File: tests/test_service.py [hunk 1]
```diff
@@ -75,6 +75,33 @@ def test_index(self):
...
+    def test_get_related_products(self):
+        """ Get related products by id tests"""
+        # Test for valid id
+        recommendation = self._create_recommendations(count=20, by_status=True)
+    
+        resp = self.app.get\
+            ("/recommendations/" + str(recommendation[0][0].id))
+
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(isinstance(resp, object), True, "Received incorrect recommendation")
```
Comment (sushanthsamala) [inline tests/test_service.py:87]: Great work so far. Can you check if the returned resp.get_json() has recommendation objects that have id == recommendation[0][0].id and the number of objects is equal to the number of objects with product-id == recommendation[0][0].id created in line 81 test_service.py?

File: tests/test_service.py [hunk 2]
```diff
@@ -75,6 +75,47 @@ def test_index(self):
...
+        if recommendation[0][0].typeid == 1: 
+            self.assertEqual(resp.get_json()[0]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+        elif recommendation[0][0].typeid == 2: 
+            self.assertEqual(resp.get_json()[1]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+        else: 
+            self.assertEqual(resp.get_json()[2]["ids"][0], recommendation[0][0].rel_id, "Received incorrect records")
+
+        num_records = len(resp.get_json()[0]["ids"]) + len(resp.get_json()[1]["ids"]) + len(resp.get_json()[2]["ids"])
+
+        self.assertEqual(num_records, 1, "Received incorrect records")
```
Comment (sushanthsamala) [inline tests/test_service.py:98]: I'm wondering why the num_records should be equal to 1?

**Review by sushanthsamala — CHANGES_REQUESTED** [review]

File: tests/test_service.py (hunk 2 above)
Comment (lj894) [reply tests/test_service.py:98]: I saw in RecommendationFactory, a new id is generated every time the class initialized:  `id = FuzzyChoice(choices=range(1, 10001))`. So when the code calls `self._create_recommendations(count=20, by_status=True)`. it creates 20 records with different ids. Therefore when you use the API to query a given id, only 1 matching record can be found among one of the 3 relationship_types.

Comment (sushanthsamala) [reply tests/test_service.py:98]: Oh okay makes sense

**Review by sushanthsamala — APPROVED** [review]

**Pull request merged**